main.nf

```diff
@@ -3,50 +3,56 @@
 
 // check parameters
 if (!params.inputDir) {
-    exit 1, "inputDir parameter is missing."
-}
-if (!params.outputDir) {
-    exit 1, "outputDir parameter is missing."
+    exit 1, 'inputDir parameter is missing.'
 }
 if ( !new File(params.inputDir).exists()) {
     exit 1, "The input folder does not exists."+params.inputDir+"\n"
 }
-if (!params.outPrefix) {
-    exit 1, "Output prefix parameter is missing."
+if (!params.outputDir) {
+    exit 1, "outputDir parameter is missing."
 }
-if (!params.debug) {
-    exit 1, "Debug prefix parameter is missing."
+if (!params.starTmpDir) {
+    exit 1, "starTmpDir parameter is missing."
+}
+
+if ( !new File(params.starTmpDir).exists()) {
+    exit 1, 'The STAR temporary folder does not exists. ('+params.starTmpDir+')\n'
 }
 if (!params.localOutDir) {
     params.localOutDir='bam'
 }
+if (!params.debug) {
+    exit 1, "Debug prefix parameter is missing."
+}
+
+
+pairFiles_ch = Channel.fromFilePairs( params.inputDir+"/"+params.inputPattern, size: 2, checkIfExists: true )
 
-pairFiles_ch = Channel.fromFilePairs( params.inputDir+"/*{1,2}.fastq.gz", size: 2, checkIfExists: true )
 
 process ERValign {
     tag "${sample}"
     
     // executor configuration
     time '8h'
-    // memory '35.GB'
+    memory '35.GB'
     scratch true
     cpus params.cpus
-    storeDir params.outputDir
+    publishDir params.outputDir
 
     // other configuration
     echo true
     errorStrategy 'terminate'
 
     input:
-    val(outPrefix) from params.outPrefix
+    tuple val(sample), file(reads) from pairFiles_ch
     val(localOutDir) from params.localOutDir
     val(limitMemory) from params.limitMemory
     val(debug) from params.debug
-    tuple val(sample), file(reads) from pairFiles_ch
     
     output:
-    path ( "${localOutDir}/${outPrefix}Aligned.sortedByCoord.out.bam" ) into bam_ch
-    path ( "${localOutDir}/${outPrefix}Aligned.sortedByCoord.out.bam.bai" ) into bai_ch
+    path ( "${localOutDir}/${sample}.Aligned.sortedByCoord.out.bam" ) into bam_ch
+    path ( "${localOutDir}/${sample}.Aligned.sortedByCoord.out.bam.bai" ) into bai_ch
+    val "${sample}" into prefix_ch
```

Comment:
Again, it's just a matter of style. Parentheses are not mandatory in groovy, but I'd either use them (+1 for me) or not.
Here you could just use val (sample) into prefix_ch